tests/E2E Tests/TokenAcquirerTests/TokenAcquirer.cs

```diff
@@ -299,5 +301,45 @@ private static async Task CreateGraphClientAndAssert(TokenAcquirerFactory tokenA
             Assert.NotNull(result.AccessToken);
         }
     }
+
+    public class AcquireTokenManagedIdentity 
+    {
+        [OnlyOnAzureDevopsFact]
+        //[Fact]
+        public async Task AcquireTokenWithManagedIdentity_UserAssigned()
+        {
+            const string scope = "https://vault.azure.net/.default";
+            const string baseUrl = "https://vault.azure.net";
+            const string clientId = "9c5896db-a74a-4b1a-a259-74c5080a3a6a";
+            TokenAcquirerFactory tokenAcquirerFactory = TokenAcquirerFactory.GetDefaultInstance();
+            _ = tokenAcquirerFactory.Services;
+            IServiceProvider serviceProvider = tokenAcquirerFactory.Build();
+
+            // Get the authorization header provider and add the options to tell it to use Managed Identity
+            IAuthorizationHeaderProvider? api = serviceProvider.GetRequiredService<IAuthorizationHeaderProvider>();
+            Assert.NotNull(api);
+            string result = await api.CreateAuthorizationHeaderForAppAsync(scope, GetAuthHeaderOptions_ManagedId(baseUrl, clientId));
```

Comment:
>GetAuthHeaderOptions_ManagedId [](http://example.com/codeflow?start=82&length=30)

when we write up the docs, customers will probably just set the options directly, so keep that in mind (not create another method to call).

Reply:
@jennyf19, I'm not sure what the ask is here... I thought customers would have these values from the appsettings.json config and not need to create their own options like I do here in the test and the method is essentially just to mock that config since I don't have an appsettings file to reference.

Reply:
Both are possible:
- customers can set it in the appsettings.json (in the DowntreamApis section)
-  but this is overridable by code (and therefore settable by code)